cpp/src/NVCategory.cu

```diff
@@ -318,11 +334,9 @@ NVCategory* NVCategory::create_from_array(const char** strs, unsigned int count)
     if( count==0 )
         return rtn;
     NVStrings* dstrs = NVStrings::create_from_array(strs,count);
-    std::pair<const char*,size_t>* indexes = nullptr;
-    RMM_ALLOC(&indexes, count * sizeof(std::pair<const char*,size_t>),0);
-    dstrs->create_index(indexes);
-    NVCategoryImpl_init(rtn->pImpl,indexes,count,true,true);
-    RMM_FREE(indexes,0);
+    rmm::device_vector<std::pair<const char*,size_t>> indexes(count);
```

Comment:
I've run into trouble making a device vector of `std::pair` before since you're using what is technically a host-only object in device code. Better to use `thrust::pair`.

Reply:
Thanks. Unfortunately, this is not a trivial change and has been working well since the original release.